src/maxdiffusion/models/train.py

```diff
@@ -522,6 +523,22 @@ def compute_loss(unet_params):
         mllog_utils.maybe_train_step_log(config, start_step, step_num, samples_count, train_metric)
     max_utils.close_summary_writer(writer)
 
+    del pipeline
+    del params
+    del unet_state
+    del vae_state
+
```

Comment:
LGTM. And feel free to merge in.

Just curious how garbage/memory collection is performed in tpu side. We can do something like `gc.collect()` in cpu. Not a problem as long as we don't hit the hbm limit.

Reply:
I didn't see any issues with memory but if we run into it, there is a function in max_utils, `delete_pytree` that can explicitly delete memory allocations on TPU.